src/lib/orionld/common/orionldEntitiyPayloadCheck.cpp

```diff
@@ -0,0 +1,261 @@
+  {
+    OBJECT_CHECK(orionldState.requestTree, "toplevel");
+    //
+    // Check presence of mandatory fields
+    //
+    if (checkEntityIdFieldExists() == false) return false;
+  }
+  char*    detailsP;
+  KjNode*  locationNodeP          = NULL;
+  KjNode*  observationSpaceNodeP  = NULL;
+  KjNode*  operationSpaceNodeP    = NULL;
+  KjNode*  createdAtP             = NULL;
+  KjNode*  modifiedAtP            = NULL;
+
+  //
+  // Check for duplicated items and that data types are correct
+  //
+  while (kNodeP != NULL)
+  {
+    if (isBatchOperation == true)
+    {
+      if (strcmp(kNodeP->name, "id") == 0)
+      {
+        orionldState.payloadIdNode = kNodeP;
+        if ((urlCheck(orionldState.payloadIdNode->value.s, &detailsP) == false) && (urnCheck(orionldState.payloadIdNode->value.s, &detailsP) == false))
+        {
+          orionldErrorResponseCreate(OrionldBadRequestData, "Invalid Entity id", "The id specified cannot be resolved to a URL or URN", OrionldDetailString);
+          return false;
+        }
+        //
+        // If the entity already exists, an error should be returned
+        //
+        // if (mongoEntityExists(orionldState.payloadIdNode->value.s, orionldState.tenant) == true)
+        // {
+        //   orionldErrorResponseCreate(OrionldAlreadyExists, "Entity already exists", orionldState.payloadIdNode->value.s, OrionldDetailsString);
+        //   ciP->httpStatusCode = SccConflict;
+        //   return false;
+        // }
+        LM_TMP(("ID: %s", orionldState.payloadIdNode->value.s));
+      }
+
+      if (strcmp(kNodeP->name, "type") == 0)
+      {
+        orionldState.payloadTypeNode = kNodeP;
+        LM_TMP(("Type: %s", orionldState.payloadTypeNode->value.s));
+      }
+    }
+
+    if (SCOMPARE9(kNodeP->name, 'l', 'o', 'c', 'a', 't', 'i', 'o', 'n', 0))
+    {
+      DUPLICATE_CHECK(locationNodeP, "location", kNodeP);
+      // FIXME: check validity of location - GeoProperty
+    }
+    else if (SCOMPARE17(kNodeP->name, 'o', 'b', 's', 'e', 'r', 'v', 'a', 't', 'i', 'o', 'n', 'S', 'p', 'a', 'c', 'e', 0))
+    {
+      DUPLICATE_CHECK(observationSpaceNodeP, "observationSpace", kNodeP);
+      // FIXME: check validity of observationSpace - GeoProperty
+    }
+    else if (SCOMPARE15(kNodeP->name, 'o', 'p', 'e', 'r', 'a', 't', 'i', 'o', 'n', 'S', 'p', 'a', 'c', 'e', 0))
+    {
+      DUPLICATE_CHECK(operationSpaceNodeP, "operationSpace", kNodeP);
+      // FIXME: check validity of operationSpaceP - GeoProperty
+    }
+    else if (SCOMPARE10(kNodeP->name, 'c', 'r', 'e', 'a', 't', 'e', 'd', 'A', 't', 0))
+    {
+      DUPLICATE_CHECK(createdAtP, "createdAt", kNodeP);
+      STRING_CHECK(kNodeP, "createdAt");
+    }
+    else if (SCOMPARE11(kNodeP->name, 'm', 'o', 'd', 'i', 'f', 'i', 'e', 'd', 'A', 't', 0))
+    {
+      DUPLICATE_CHECK(modifiedAtP, "modifiedAt", kNodeP);
+      STRING_CHECK(kNodeP, "modifiedAt");
+    }
+    else  // Property/Relationshiop - must check chars in the name of the attribute
```

Comment:
So, do it.
Always use the wiord FIXME for this kind of comments.
In case we forget it, we'll always find it when grepping for FIXME.

There should be a function to validate attribute names. If we don't have one, we should create it